server/api.go

```diff
@@ -1,22 +1,31 @@
 package server
 
 import (
+	"context"
 	"net/http"
+	"time"
 
 	"github.com/Hukyl/genesis-kma-school-entry/models"
 	"github.com/gin-gonic/gin"
 )
 
+const (
+	RatePath      = "/rate"
+	SubscribePath = "/subscribe"
+)
+
 type UserRepository interface {
 	Exists(user *models.User) (bool, error)
 	Create(user *models.User) error
 }
 
 // NewGetRateHandler is a handler that fetches the exchange rate between USD and UAH
 // from a RateFetcher interface and returns it as a JSON response.
-func NewGetRateHandler(rateFetcher RateFetcher) func(*gin.Context) {
+func NewGetRateHandler(rateService RateService, timeout time.Duration) func(*gin.Context) {
 	return func(c *gin.Context) {
-		rate, err := rateFetcher.FetchRate("USD", "UAH")
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		rate, err := rateService.FetchRate(ctx, "USD", "UAH")
```

Comment:
btw, now these currency codes are used also in your tests :) so, maybe it's time to extract them to constants?

Reply:
it could be even local constants to this package, why not?